File: src/entity/evenement.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package entity;

import java.sql.Date;
import java.time.LocalDate;



/**
 *
 * @author zakar
 */
public class evenement {
    
    private int id_event;
    private String nom_event ;
    private String emplacement;
    private LocalDate date_debut;
    private LocalDate date_fin;
    private String description_event;
    private int prix;
    private String affiche;
    private int nbr_max ;
    private user id_user; 

 
    public evenement(int id_event, String nom_event1, String emplacement1, LocalDate date_d, LocalDate date_f, String description, int prix2, String image, int nbr_max1, int id_user1) {
        this.id_event = id_event;
    }

    
    
    public evenement(int id_event, String nom_event, String emplacement, LocalDate date_debut, LocalDate date_fin, String description_event, int prix, int nbr_max, user id_user) {
        this.id_event = id_event;
        this.nom_event = nom_event;
        this.emplacement = emplacement;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description_event = description_event;
        this.prix = prix;
        this.nbr_max = nbr_max;
        this.id_user = id_user;
    }

    
    
    
    public evenement(String nom_event, String emplacement, LocalDate date_debut, LocalDate date_fin, String description_event, int prix, String affiche, int nbr_max, user id_user) {
        this.nom_event = nom_event;
        this.emplacement = emplacement;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description_event = description_event;
        this.prix = prix;
        this.affiche = affiche;
        this.nbr_max = nbr_max;
        this.id_user = id_user;
    }

    public evenement() {
    }

    public evenement(int id_event, String nom_event, String emplacement, LocalDate date_debut, LocalDate date_fin, String description_event, int prix, String affiche, int nbr_max, user id_user) {
        this.id_event = id_event;
        this.nom_event = nom_event;
        this.emplacement = emplacement;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description_event = description_event;
        this.prix = prix;
        this.affiche = affiche;
        this.nbr_max = nbr_max;
        this.id_user = id_user;
    }

    public int getNbr_max() {
        return nbr_max;
    }

    public void setNbr_max(int nbr_max) {
        this.nbr_max = nbr_max;
    }

    public evenement(int id_event, String nom_event, String emplacement, LocalDate date_debut, LocalDate date_fin, String description_event, int prix, user id_user) {
        this.id_event = id_event;
        this.nom_event = nom_event;
        this.emplacement = emplacement;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description_event = description_event;
        this.prix = prix;
        this.id_user = id_user;
    }

    public evenement(int id_event, String nom_event, String emplacement, LocalDate date_debut, LocalDate date_fin, String description_event, int prix, String affiche, user id_user) {
        this.id_event = id_event;
        this.nom_event = nom_event;
        this.emplacement = emplacement;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description_event = description_event;
        this.prix = prix;
        this.affiche = affiche;
        this.id_user = id_user;
    }

    public evenement( String nom_event, String emplacement, LocalDate date_debut, LocalDate date_fin, String description_event, int prix, user id_user) {
        this.nom_event = nom_event;
        this.emplacement = emplacement;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description_event = description_event;
        this.prix = prix;
        this.id_user = id_user;
    }

    public evenement(String nom_event, String emplacement, LocalDate date_debut, LocalDate date_fin, String description_event, int prix, String affiche, user id_user) {
        this.nom_event = nom_event;
        this.emplacement = emplacement;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.description_event = description_event;
        this.prix = prix;
        this.affiche = affiche;
        this.id_user = id_user;
    }

    public String getAffiche() {
        return affiche;
    }

    public void setAffiche(String affiche) {
        this.affiche = affiche;
    }

    public user getId_user() {
        return id_user;
    }

    public void setId_user(user id_user) {
        this.id_user = id_user;
    }

   

    public int getId_event() {
        return id_event;
    }

    public String getNom_event() {
        return nom_event;
    }

    public String getEmplacement() {
        return emplacement;
    }

    public LocalDate getDate_debut() {
        return date_debut;
    }

    public LocalDate getDate_fin() {
        return date_fin;
    }

    public String getDescription_event() {
        return description_event;
    }

    public int getPrix() {
        return prix;
    }

    public void setId_event(int id_event) {
        this.id_event = id_event;
    }

    public void setNom_event(String nom_event) {
        this.nom_event = nom_event;
    }

    public void setEmplacement(String emplacement) {
        this.emplacement = emplacement;
    }

    public void setDate_debut(LocalDate date_debut) {
        this.date_debut = date_debut;
    }

    public void setDate_fin(LocalDate date_fin) {
        this.date_fin = date_fin;
    }

    public void setDescription_event(String description_event) {
        this.description_event = description_event;
    }

    public void setPrix(int prix) {
        this.prix = prix;
    }
     public void setUser(user u) {
    this.id_user = u;
}

    @Override
    public String toString() {
        return "evenement{" + "id_event=" + id_event + ", nom_event=" + nom_event + ", emplacement=" + emplacement + ", date_debut=" + date_debut + ", date_fin=" + date_fin + ", description_event=" + description_event + ", prix=" + prix + ", affiche=" + affiche + ", nbr_max=" + nbr_max + ", id_user=" + id_user + '}';
    }

 

 
    
    
    
    
    
}

File: src/Entity/Evenement.php
<?php

namespace App\Entity;

use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Validator\Context\ExecutionContextInterface;
use Symfony\Component\Serializer\Annotation\Groups;



/**
 * Evenement
 *
 * @ORM\Table(name="evenement", indexes={@ORM\Index(name="id_user_idx", columns={"id_user"})})
 * @ORM\Entity
 */
class Evenement
{
    #[Groups("events")]
    /**
     * @var int
     *
     * @ORM\Column(name="id_event", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     * 
     */
    private $idEvent;

    #[Groups("events")]
    /**
     * @var string
     *
     * @ORM\Column(name="nom_event", type="string", length=45, nullable=false)
     * 
     */
    private $nomEvent;
    
    #[Groups("events")]
    /**
     * @var string
     *
     * @ORM\Column(name="emplacement", type="string", length=45, nullable=false)
     * 
     */
    private $emplacement;
    
   
    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_debut", type="date", nullable=false)
     * 
     */
    private $dateDebut;
 
    
    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_fin", type="date", nullable=false)
     * 
     */
    private $dateFin;
    
    #[Groups("events")]

    /**
     * @var string
     *
     * @ORM\Column(name="description_event", type="string", length=45, nullable=false)
     *
     */
    private $descriptionEvent;

    #[Groups("events")]
    /**
     * @var int
     *
     * @ORM\Column(name="prix", type="integer", nullable=false)
     * 
     */
    private $prix;

    #[Groups("events")]
    /**
     * @var string|null
     *
     * @ORM\Column(name="affiche", type="string", length=45, nullable=true)
     * 
     */
    private $affiche;

    #[Groups("events")]
    /**
     * @var int|null
     *
     * @ORM\Column(name="nbr_max", type="integer", nullable=true)
     * 
     */
    private $nbrMax;

    /**
     * @var int
     *
     * @ORM\Column(name="NbrView", type="integer", nullable=false)
     * 
     */
    private $NbrView;

    /**
     * @var string
     *
     * @ORM\Column(name="random_color", length=45, nullable=false)
     * 
     */
    private $randomColor;

    /**
     * @var \User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_user", referencedColumnName="id_user")
     *
     * })
     */
    private $idUser;

    public function getIdEvent(): ?int
    {
        return $this->idEvent;
    }

    public function getNomEvent(): ?string
    {
        return $this->nomEvent;
    }

    public function setNomEvent(string $nomEvent): self
    {
        $this->nomEvent = $nomEvent;

        return $this;
    }

    public function getEmplacement(): ?string
    {
        return $this->emplacement;
    }

    public function setEmplacement(string $emplacement): self
    {
        $this->emplacement = $emplacement;

        return $this;
    }

    public function getDateDebut(): ?\DateTimeInterface
    {
        return $this->dateDebut;
    }

    public function setDateDebut(\DateTimeInterface $dateDebut): self
    {
        $this->dateDebut = $dateDebut;

        return $this;
    }

    public function getDateFin(): ?\DateTimeInterface
    {
        return $this->dateFin;
    }

    public function setDateFin(\DateTimeInterface $dateFin): self
    {
        $this->dateFin = $dateFin;

        return $this;
    }

    public function getDescriptionEvent(): ?string
    {
        return $this->descriptionEvent;
    }

    public function setDescriptionEvent(string $descriptionEvent): self
    {
        $this->descriptionEvent = $descriptionEvent;

        return $this;
    }

    public function getPrix(): ?int
    {
        return $this->prix;
    }

    public function setPrix(int $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getAffiche(): ?string
    {
        return $this->affiche;
    }

    public function setAffiche(?string $affiche): self
    {
        $this->affiche = $affiche;

        return $this;
    }

    public function getNbrMax(): ?int
    {
        return $this->nbrMax;
    }

    public function setNbrMax(?int $nbrMax): self
    {
        $this->nbrMax = $nbrMax;

        return $this;
    }
    public function getNbrView(): ?int
    {
        return $this->NbrView;
    }

    public function setNbrView(int $NbrView): self
    {
        $this->NbrView = $NbrView;

        return $this;
    }
    public function getRandom(): ?string
    {
        return $this->randomColor;
    }

    public function setRandom(string $randomColor): self
    {
        $this->randomColor = $randomColor;

        return $this;
    }
    public function getIdUser(): ?User
    {
        return $this->idUser;
    }

    public function setIdUser(?User $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }
/**
     * @Assert\Callback
     */
    public function validateDates(ExecutionContextInterface $context, $payload)
    {
        if ($this->getDateFin() < $this->getDateDebut()) {
            $context->buildViolation('La date de début doit être antérieure à la date de fin')
                ->atPath('dateDebut')
                ->addViolation();
        }
    }

}